priv/codegen/templates/_schema.eex

```diff
@@ -1,11 +1,20 @@
-[
+{
   <%= for type <- @schema do %>
     <%= case type do %>
+      <% {name, [:builtin, schema]} -> %>
+        <%= name %>: %<%= @render_schema.(
+          schema: schema, 
+          render_schema: @render_schema, 
+          paths: @paths ++ [name]) 
+        %>,
       <% {name, schema} -> %>
-  <%= name %>: <%= @render_schema.(schema: schema, render_schema: @render_schema) %>,
-
+        <%= name %>: %Result.<%= Enum.map_join(@paths ++ [name],".", &(Macro.camelize(&1))) %><%= @render_schema.(
+          schema: schema, 
+          render_schema: @render_schema, 
+          paths: @paths ++ [name]) 
+        %>,
       <% name -> %>
-  :<%= name %>,
+        <%= name %>: result<%= Enum.map_join(@paths, &("[\"#{&1}\"]")) %>["<%= name %>"],
```

Comment:
It's possible that this problem also affects normal properties, not just relationships, because I noticed for the query from `test/support/codegen/edgeql/types/select_result.edgeql` the following mapping 
```elixir
mp_named_tuple_x_int64_y_int64: %{
  y: result["mp_named_tuple_x_int64_y_int64"]["y"],
  x: result["mp_named_tuple_x_int64_y_int64"]["x"]
},
```
 This should be wrapped in `Enum.map/2` for cases where it is a mutli property or an array + this code doesn't account for indexes as fields for named tuples though they are in the typespec